src/controllers/scripting/legacy/controllerscriptenginelegacy.cpp

```diff
@@ -0,0 +1,198 @@
+#include "controllers/scripting/legacy/controllerscriptenginelegacy.h"
```

Comment:
Since this is the legacy part, why is it necessary to split the functions in controllerscriptinterface.cpp and controllerscriptenginelegacy.cpp .
It is very hard to review and there should be no functional change right? 
I am afraid the testing effort and regression risk is not worth it. 
How about revert the change and just move the unchanged old files into the legacy folder.

Reply:
Because QJSEngine takes ownership of the object and the QJSEngine is a member of ControllerScriptEngineLegacy. @ferranpujolcamins stumbled upon this in #1795 and created the old ControllerEngineJSProxy class to hack around this which has this comment at the top of the header in master:
```
// An object of this class gets exposed to the JS engine, so the methods of this class
// constitute the api that is provided to scripts under "engine" object.
//
// The implementation simply forwards its method calls to the ControllerEngine.
// We cannot expose ControllerEngine directly to the JS engine because the JS engine would take
// ownership of ControllerEngine. This is problematic when we reload a script file, because we
// destroy the existing JS engine to create a new one. Then, since the JS engine owns ControllerEngine
// it will try to delete it. See this Qt bug: https://bugreports.qt.io/browse/QTBUG-41171
```
This was a quick hack around the tight coupling in ControllerEngine. This old design was a mess to work with. controllerengine.cpp in master is 1325 lines of code. Here I have moved the methods which ControllerJSProxy invoked on ControllerEngine onto the ControllerJSProxy class then renamed ControllerJSProxy to ControllerScriptInterface. If you have a suggestion for a better name than ControllerScriptInterface, please suggest it.

Reply:
If you look at the order of commits, I did this split before splitting ControllerScriptModuleEngine from ControllerScriptEngineBase. At this point it would be quite a bit of work to undo that and put the code back to a worse organization.

Reply:
>  It is very hard to review and there should be no functional change right?

Yes, it is hard to review, and yes there should be no functional change. Any refactoring this big will be hard to review.

Reply:
I did some manual diffs, and unfortunately there are some functional changes.  :-/
This means you can either revert the split, or we need some extensive review or testing. 

I am hesitating to do this for legacy code. 
We have here in one PR renaming of files, reformatting, split of files and functional changes in legacy code. 

If you really insist that all this needs to be done, please debase this PR and put all the changes in small reviewable commits and disable the clang-format hook.

Reply:
I have made an effort to keep these commits as small as practical without leaving commits with either broken functionality or inappropriately named code. Have you looked through this commit by commit?

Reply:
Sorry that fails probably because if clang-format. 
Can you do it again without it?